src/components/ContactModal.test.tsx

```diff
@@ -0,0 +1,233 @@
+import { UpsertContactModal } from "./ContactModal";
+import { contact1, contact2 } from "../mocks/contacts";
+import { mockImplicitAddress } from "../mocks/factories";
+import { fireEvent, render, screen, waitFor } from "../mocks/testUtils";
+import { contactsActions } from "../utils/redux/slices/contactsSlice";
+import { store } from "../utils/redux/store";
+
+describe("UpsertContactModal", () => {
+  describe("on adding contact", () => {
+    describe.each([
+      { testCase: "new contact", renderComponent: () => render(<UpsertContactModal />) },
```

Comment:
you don't have to use `render` here, you can just leave the element itself and use render in the test case